src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShutdownTask.java

```diff
@@ -97,47 +105,40 @@ public TaskResult call() {
         boolean applicationException = false;
 
         try {
-            ShutdownReason localReason = reason;
-            List<Shard> latestShards = null;
-            /*
-             * Revalidate if the current shard is closed before shutting down the shard consumer with reason SHARD_END
-             * If current shard is not closed, shut down the shard consumer with reason LEASE_LOST that allows active
-             * workers to contend for the lease of this shard.
-             */
-            if(localReason == ShutdownReason.TERMINATE) {
-                ShardClosureVerificationResponse shardClosureVerificationResponse = kinesisProxy.verifyShardClosure(shardInfo.getShardId());
-                if (shardClosureVerificationResponse instanceof ShardListWrappingShardClosureVerificationResponse) {
-                    latestShards = ((ShardListWrappingShardClosureVerificationResponse)shardClosureVerificationResponse).getLatestShards();
-                }
+            LOG.debug("Invoking shutdown() for shard " + shardInfo.getShardId() + ", concurrencyToken: "
+                              + shardInfo.getConcurrencyToken() + ", Shutdown reason: " + reason);
 
-                // If shard in context is not closed yet we should shut down the ShardConsumer with Zombie state
-                // which avoids checkpoint-ing with SHARD_END sequence number.
-                if(!shardClosureVerificationResponse.isShardClosed()) {
-                    localReason = ShutdownReason.ZOMBIE;
-                    dropLease();
-                    LOG.info("Forcing the lease to be lost before shutting down the consumer for Shard: " + shardInfo.getShardId());
+            // If we reached end of the shard, set sequence number to SHARD_END.
+            if (reason == ShutdownReason.TERMINATE) {
+                // Create new lease for the child shards if they don't exist.
+                // We have one valid scenario that shutdown task got created with SHARD_END reason and an empty list of childShards.
+                // This would happen when KinesisDataFetcher(for polling mode) or FanOutRecordsPublisher(for StoS mode) catches ResourceNotFound exception.
```

Comment:
Nit: since this is 1.x we only have polling mode, we can remove StoS mode from the comment

Reply:
Good Catch! Will change this comment